gobblin-runtime/src/main/java/gobblin/runtime/api/JobCatalog.java

```diff
@@ -24,7 +24,10 @@
   /** Returns an immutable {@link Collection} of {@link JobSpec}s that are known to the catalog. */
   Collection<JobSpec> getJobs();
 
-  /** Get a {@link JobSpec} by uri. */
+  /**
+   * Get a {@link JobSpec} by uri.
+   * @throws JobSpecNotFoundException if no such JobSpec exists
+   **/
   JobSpec getJobSpec(URI uri);
```

Comment:
It's a RuntimeException. I think under normal circumstances, it should rarely happen so I am not sure the API users should be burdened with always doing try-catch. What do you think?